### DISCUSSION TITLE
How do i make custom command, that works on servers (basicly client only command)

### DISCUSSION THREAD

**Discussion by AgentKosticka** [FabricMC/fabric-api#2332] [Mod Dev Support] [answered]
Title speaks for it's self. I can't figure out how to make client only command, i am running into weird issues. I just dont know how to do it. IF YOU KNOW THE ANSVER COMMENT PLEASE

**Comment by warjort** [comment]
> Title speaks for it's self. 

Not really, how can you have a client only command that works on servers? :-)

Anyway. This is the test code for the fabric client command api:
https://github.com/FabricMC/fabric/blob/1.19/fabric-command-api-v2/src/testmod/java/net/fabricmc/fabric/test/command/client/ClientCommandTest.java

Maybe it will help you do whatever it is you are trying to do?
Otherwise, you will need to be more specific.

**Reply by AgentKosticka** [reply]
I am trying to make mod, that adds more tabs with just /tab blablabla. I still can't figure it out how to make the thing work (The thing as of the commands)

**Reply by warjort** [reply]
Your question is still unanswerable.
What are "tabs"? The only tabs I can think of in vanilla minecraft are the creative mode item tabs/categories.

To get your question answered. You need to show what you have tried and why it doesn't work, e.g. error messages.
Just saying "it doesn't work" without properly defining the "it" or "work" will likely just lead to your question getting ignored.

Compare your code with the 60+ examples I posted below, or explain why none of them are useful.

**Reply by AgentKosticka** [reply]
sorry to bother you, i am just an unexperienced java developer. I will try to use those examples posted bellow to create one. Thanks (btw i am currently looking at this example: [https://github.com/Sjouwer/tp-utils/blob/1.18/src/main/java/io/github/sjouwer/tputils/TpUtils.java](TpUlits) and I cant figure out, how did he import me.shedaniel.autoconfig.blabla when it isn't in his project)

**Reply by warjort** [reply]
autoconfig is part of cloth config
https://github.com/Sjouwer/tp-utils/blob/33e772b5af416ed6c9c5d3a6dcb77bfda4f4fb0b/build.gradle#L36

**Comment by warjort** [comment, marked as answer]
These are examples of people using the registration callback on github
https://github.com/search?q=ClientCommandRegistrationCallback&type=code
